iterative/resource_runner.go

```diff
@@ -438,8 +438,13 @@ func isAMIAvailable(cloud string, region string) bool {
 		regions = azure.ImageRegions
 	}
 
+	var regionCloud = aws.GetRegion(region)
+	if cloud == "azure" {
+		regionCloud = azure.GetRegion(region)
+	}
```

Comment:
```suggestion
	var regionCloud string
	switch cloud {
	case "aws":
		regionCloud = aws.GetRegion(region)
	case "azure":
		regionCloud = azure.GetRegion(region)
	}
```

Can we use a `switch` structure so it's cleaner once we add more vendors?

Reply:
sure! I never use them because I hate them. In many languages is so missused and so error prone that  I almost removed it from my voc

Reply:
If you prefer, we can keep extending it with chained conditionals. I'm open to any of the approaches.

Go's `switch` statement is pretty sane in my opinion, and doesn't [`fallthrough`](https://github.com/golang/go/wiki/Switch#fall-through) unless you explicitly require it to.

Reply:
Ideally we should write something like:


```go
var provider Provider
	switch cloud {
		case "aws":
		provider = aws
	case "azure":
		provider = azure
	}

	regions := provider.ImageRegions
	cloudRegion := provider.GetRegion(region)
	for _, item := range regions {
		if item == cloudRegion {
			return true
		}
	}

	return false
```

But I don't know how to do Interfaces in go

Reply:
:100: I have been thinking on this for a few months, but we always had something more important to do. If you want, I can open a follow-up pull request with a proposal similar to your snippet above.

Reply:
Closing this thread with 436bfa209dca1fed319ef38cd639d1d1e875bed7 and #147